File: go/main_test.go
package main

import "testing"

func TestCountPrimes(t *testing.T) {
	tests := []struct {
		limit    int
		expected int
	}{
		{limit: 0, expected: 0},
		{limit: 2, expected: 0},
		{limit: 10, expected: 4},
		{limit: 20, expected: 8},
		{limit: 100, expected: 25},
	}
	for _, tt := range tests {
		actual := CountPrimes(tt.limit)
		if actual != tt.expected {
			t.Errorf("CountPrimes(%d) expected %d, got %d", tt.limit, tt.expected, actual)
		}
	}
}

File: java/MainTest.java
public class MainTest {
    public static void main(String[] args) {
        assert Main.countPrimes(0) == 0 : "Expected 0";
        assert Main.countPrimes(2) == 0 : "Expected 0";
        assert Main.countPrimes(10) == 4 : "Expected 4 but got " + Main.countPrimes(10);
        assert Main.countPrimes(20) == 8 : "Expected 8";
        assert Main.countPrimes(100) == 25 : "Expected 25";
        System.out.println("Java tests passed.");
    }
}
